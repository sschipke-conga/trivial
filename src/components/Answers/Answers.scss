.answers {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
}

.answer {
  background: rgba(0, 0, 0, 0.2);
  color: white;
  font-size: 1.3em;
  height: 30px;
  margin: 10px;
  min-width: 300px;
  &:hover {
    background: black;
    outline: 3px solid white;
  }
}