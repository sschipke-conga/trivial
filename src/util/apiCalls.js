export const fetchQuestions = async (amount, category, difficulty) => {
  let url = `https://opentdb.com/api.php?amount=${amount}&category=${category}&difficulty=${difficulty}&type=multiple`
  const res = await fetch(url);
  if (!res.ok) {
    throw Error('Failed to fetch')
  }
  const data = await res.json()
  return data.results
}
