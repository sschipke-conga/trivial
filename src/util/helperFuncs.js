export const shuffle = answers => {
  var m = answers.length, t, i;
  while (m) {
    i = Math.floor(Math.random() * m--);
    t = answers[m];
    answers[m] = answers[i];
    answers[i] = t;
  }
  return answers;
}

export const determineCurrentTeam = (turnCount) => {
  if (turnCount % 2 === 0) {
    return 1
  } 
  return 0
}

export const determineWinner = teams => {
  if(teams[0].score === teams[1].score) {
    return false
  }
  return teams.sort((teamOne, teamTwo) => {
    return teamTwo.score-teamOne.score
  })[0]
}

export const capDifficulty = difficulty => {
  return difficulty[0].toUpperCase() + difficulty.slice(1)
}

export const cleanEncoding = (string) => {
  let cleanPhrase = string.replace(/&#039;/g, "'")
  .replace(/&quot;/g, '"')
  .replace(/&eacute/g, 'Ã©');
  return cleanPhrase
}

export const cleanQuestions = (questions) => {
    return questions.map(question => {
    question.question = cleanEncoding(question.question)
    question.correct_answer = cleanEncoding(question.correct_answer)
    question.incorrect_answers = question.incorrect_answers.map(answer => cleanEncoding(answer))
    return question
    })
}