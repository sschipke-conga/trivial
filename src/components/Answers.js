import React from 'react';
import {shuffle} from '../util/helperFuncs';

const Answers = ({answers}) => {
  const shuffledAnswers = shuffle(answers).map((answer, i) => 
    <button key={i} type="button" className="answer">{answer}</button>
    )
  return (
    <section className="answers">
      {shuffledAnswers}
    </section>
  )
}

export default Answers