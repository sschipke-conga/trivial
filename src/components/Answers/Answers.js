import React from 'react';
import {shuffle} from '../../util/helperFuncs';
import './Answers.scss';

const Answers = ({answers, submitAnswer}) => {
  const shuffledAnswers = shuffle(answers).map((answer, i) => 
    <button onClick={submitAnswer} key={i} name={answer} type="button" className="answer">{answer}</button>
    )
  return (
    <section className="answers">
      {shuffledAnswers}
    </section>
  )
}

export default Answers