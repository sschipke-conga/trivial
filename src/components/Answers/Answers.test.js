import React from 'react';
import { shallow } from 'enzyme';
import { shuffle } from '../../util/helperFuncs';
import Answers from './Answers';

jest.mock('../../util/helperFuncs')

const mockAnswers = ['c', 'b', 'd', 'a']

shuffle.mockImplementation(() => ['b', 'd', 'a', 'c'])

describe('Answers', () => {
  let wrapper;
  const mockSubmitAnswer = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<Answers 
    answers={mockAnswers}
    submitAnswer={mockSubmitAnswer}
    />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should call shuffle when mounted', () => {
    expect(shuffle).toHaveBeenCalledWith(mockAnswers)
  })
})