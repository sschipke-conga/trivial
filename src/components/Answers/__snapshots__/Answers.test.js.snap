// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answers should match the snapshot 1`] = `
<section
  className="answers"
>
  <button
    className="answer"
    key="0"
    name="b"
    onClick={[MockFunction]}
    type="button"
  >
    b
  </button>
  <button
    className="answer"
    key="1"
    name="d"
    onClick={[MockFunction]}
    type="button"
  >
    d
  </button>
  <button
    className="answer"
    key="2"
    name="a"
    onClick={[MockFunction]}
    type="button"
  >
    a
  </button>
  <button
    className="answer"
    key="3"
    name="c"
    onClick={[MockFunction]}
    type="button"
  >
    c
  </button>
</section>
`;
